### PR TITLE
Request level coordinator slow logs

### PR DESCRIPTION
**Pull request by dzane17**
### Description
Draft

```
// Updated log example
[2023-10-12T17:29:58,151][INFO ][c.s.r.s.t.SearchRequestSlowLog] [runTask-0] took[6ms], took_millis[6], phase_took_millis[{expand=0, query=5, fetch=0}], search_type[QUERY_THEN_FETCH], total_shards[10], source[{"query":{"query_string":{"query":"abcd","fields":[],"type":"best_fields","default_operator":"or","max_determinized_states":10000,"enable_position_increments":true,"fuzziness":"AUTO","fuzzy_prefix_length":0,"fuzzy_max_expansions":50,"phrase_slop":0,"escape":false,"auto_generate_synonyms_phrase_query":true,"fuzzy_transpositions":true,"boost":1.0}}}], id[]
```

### CODE REVIEW

File: server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+    public static final Setting<TimeValue> CLUSTER_SEARCH_REQUEST_SLOWLOG_THRESHOLD_FETCH_TRACE_SETTING = Setting.timeSetting(
+        CLUSTER_SEARCH_REQUEST_SLOWLOG_PREFIX + ".fetch.trace",
+        TimeValue.timeValueNanos(-1),
+        TimeValue.timeValueMillis(-1),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
```
Comment (jainankitk) [inline server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java]: Does not make too much sense to have phase breakdown settings at the cluster level. Better will be to print the latency of different phases in the overall slow log message

Comment (jainankitk) [reply server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java]: At shard it is necessary to have the phase breakdown settings as the correlation between query and fetch phase is lost. But, at the coordinator node we have the luxury of not losing that correlation

Comment (dzane17) [reply server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java]: Ok I see what you are saying about shard slowlog correlation. I can edit the overall slowlog message to include total and individual phase times. Just need to update the listener to store the values.

We intended to allow users to set slowlog thresholds on query and fetch in addition to overall. You are saying we don't need this flexibility? 

@sgup432 @kkhatua

Comment (jainankitk) [reply server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java]: Yes, unnecessary flexibility generally causes confusion. I want end user to have single setting that directly correlates with their SLAs. And log message from the overall setting should give the breakdown across different phases.

Why will user want to set threshold on individual query or fetch phase? As I said earlier, it was necessity at shard level due to missing correlation between the phases.

Comment (sgup432) [reply server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java]: Kind of agree with @jainankitk. Its more useful for users to set a request level threshold and have phase level breakdown as part of slowlogs.

Comment (dzane17) [reply server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java]: Done
1. Removed query and fetch phase logging thresholds
2. Added phase took time to logs (sample in PR description)

**Pull request closed without merging**